lib/iris/cube.py

```diff
@@ -1624,17 +1624,31 @@ def shape(self):
 
     @property
     def dtype(self):
-        if self._dtype is None:
-            result = self.core_data.dtype
-        else:
-            result = self._dtype
-        return result
+        dtype = self._dtype
+        if dtype is None:
+            dtype = self.core_data.dtype
+        return dtype
 
     @dtype.setter
     def dtype(self, dtype):
         self._dtype = dtype
 
     @property
+    def fill_value(self):
+        fill_value = self._fill_value
+        if fill_value is not None:
+            try:
+                fill_value = np.asarray([self._fill_value],
+                                        dtype=self.dtype)[0]
+            except OverflowError:
+                fill_value = None
```

Comment:
Might be worth some sort of note to the user as to _why_ they're getting `None` as the cube's dtype in such a case.

I think it hits plausible use-cases though :+1: